pkg/infra/baseagent.go

```diff
@@ -222,5 +249,19 @@ func (a *Base) HandleResponse(msg messages.BoolResponseMessage)
 func (a *Base) HandleStateFoodTaken(msg messages.StateFoodTakenMessage)                  {}
 func (a *Base) HandleStateHP(msg messages.StateHPMessage)                                {}
 func (a *Base) HandleStateIntendedFoodTaken(msg messages.StateIntendedFoodIntakeMessage) {}
-func (a *Base) HandleProposeTreaty(msg messages.ProposeTreatyMessage)                    {}
-func (a *Base) HandleTreatyResponse(msg messages.TreatyResponseMessage)                  {}
+
+func (a *Base) HandleProposeTreaty(msg messages.ProposeTreatyMessage) {
+	treatyProposal := msg.Treaty()
+	// always add treaty to active list if proposing
+	a.AddToActiveTreaties(treatyProposal)
+}
+
+func (a *Base) HandleTreatyResponse(msg messages.TreatyResponseMessage) {
+	repliedTreaty := a.activeTreaties[msg.TreatyID()]
+	// conditionally remove treaty from active treaties once response is received
+	if msg.Response() {
+		repliedTreaty.SignTreaty()
+	} else {
+		a.DeleteFromActiveTreaties(repliedTreaty)
+	}
+}
```

Comment:
This is incorrect...? At least we shouldn't `DeleteFromActiveTreaties` if the message got rejected. We don't know if this Treaty has been also sent in the opposite direction and been agreed to.

Reply:
Treaties are always added to the bank if proposed. If the response is negative, we remove the treaty from the active list?

Reply:
> This is incorrect...? At least we shouldn't `DeleteFromActiveTreaties` if the message got rejected. We don't know if this Treaty has been also sent in the opposite direction and been agreed to.

This is the same thing as asking someone to renew a treaty. If I ask you to sign a treaty and you say yes, then you send the same one back and I decline, then you effectively have a treaty lasting 1 tick that instantly gets cancelled